src/web_service/cookie.h

```diff
@@ -0,0 +1,98 @@
+// SPDX-FileCopyrightText: 2023 Yuzu Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <ctime>
+#include <regex>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace WebService {
+
+class Cookie {
+public:
+    explicit Cookie(std::string name, std::string value);
+    Cookie() = default;
+    ~Cookie() = default;
+
+    enum class RawForm { NameAndValueOnly, Full };
+    enum class SameSite { Default, None, Lax, Strict };
+
+    static constexpr bool IsTerminator(char c) {
+        return c == '\n' || c == '\r';
+    }
+
+    static constexpr bool IsValueSeparator(char c) {
+        return IsTerminator(c) || c == ';';
+    }
+
+    static constexpr bool IsWS(char c) {
+        return c == ' ' || c == '\t' || c == '\r' || c == '\n';
+    }
+
+    static size_t AdvanceTo(std::string_view text, size_t from) {
+        // RFC 2616 defines linear whitespace as:
+        //  LWS = [CRLF] 1*( SP | HT )
+        // We ignore the fact that CRLF must come as a pair at this point
+        // It's an invalid HTTP header if that happens.
+        while (from < text.size()) {
+            if (IsWS(text[from]))
+                ++from;
+            else
+                return from; // non-whitespace
+        }
+
+        // reached the end
+        return text.size();
+    }
+
+    bool IsSecure() const;
+    void SetSecure(bool enable);
+    bool IsHttpOnly() const;
+    void SetHttpOnly(bool enable);
+    SameSite SameSitePolicy() const;
+    void SetSameSitePolicy(SameSite sameSite);
```

Comment:
```suggestion
    void SetSameSitePolicy(SameSite same_site);
```

Ditto for other camelCased names below